### PR TITLE
[dart] Default value is only available when nullable is not set

### PR DESCRIPTION
**Pull request by agilob**
Dart json_serializable does not allow setting `nullable: false` AND `defaultValue` at the same time.

As `nullable: true` with default value doesn't make sense, neither `nullable: false` it requires removal of `nullable`.

It throws exception:

```
Error with `@JsonKey` on `mapProperty`. Cannot use `defaultValue` on a field with `nullable` false.
package:openapi/model/additional_properties_class.dart:31:23
   ╷
31 │   Map<String, String> mapProperty;
   │                       ^^^^^^^^^^^

```

@kuhnroyal

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/dart2/class.mustache [hunk 1]
```diff
@@ -50,7 +50,7 @@ class {{{classname}}} {
   {{/isEnum}}
   {{^isBinary}}
   @JsonKey(
-    nullable: {{#nullable}}true{{/nullable}}{{^nullable}}false{{/nullable}},
+    {{#defaultValue}}defaultValue: {{{defaultValue}}},{{/defaultValue}}{{^defaultValue}}nullable: {{#nullable}}true,{{/nullable}}{{^nullable}}false,{{/nullable}}{{/defaultValue}}
```
Comment (kuhnroyal) [inline modules/openapi-generator/src/main/resources/dart2/class.mustache:53]: I noticed a problem earlier today. It's not `{{nullable}}` - it is `{{isNullable}}`.
And then you can shorten this to `nullable: {{isNullable}}`

Comment (agilob) [reply modules/openapi-generator/src/main/resources/dart2/class.mustache:53]: aa that's why it was behaving weird sometimes!

Comment (kuhnroyal) [reply modules/openapi-generator/src/main/resources/dart2/class.mustache]: Ya found some places in other templates that had wrong output when I tried some NNBD stuff.

**Review by kuhnroyal — APPROVED** [review]

**Comment by agilob** [comment]
@kuhnroyal bump
